feat: add data and deadline to buyCoverWithRi signature

MilGard91 opened the pull request:
## Description

Describe the changes made in this PR. A link to the issue is enough if it contains all the relevant information.

## Testing

Explain how you tested your changes to ensure they work as expected.

## Checklist

- [ ] Performed a self-review of my own code
- [ ] Made corresponding changes to the documentation

MilGard91 committed:
feat: add data and deadline to buyCoverWithRi signature

MilGard91 committed:
fix: use encoded data for the buyCoverWithRi method

MilGard91 committed:
fix: add cover ID to the event

MilGard91 committed:
feat: add ri premium and payment asset to CoverRiAllocated event

MilGard91 committed:
fix: audit findings fix

MilGard91 committed:
chore: release docs

rackstar commented on contracts/modules/cover/Cover.sol:
```diff
@@ -199,6 +204,9 @@ contract Cover is ICover, EIP712, RegistryAware, ReentrancyGuard, Multicall {
       riRequest.premium,
       params.period,
       params.coverAsset,
+      keccak256(riRequest.data),
+      riRequest.dataFormat,
+      riRequest.deadline,
```
noticed that we're not including params.owner on the signed message. is this a deliberate design decision to allow anyone to execute the quote?

rackstar commented on contracts/modules/cover/Cover.sol:
```diff
@@ -189,6 +191,9 @@ contract Cover is ICover, EIP712, RegistryAware, ReentrancyGuard, Multicall {
+          "uint32 deadline,",
           "uint256 nonce)"
         )
       ),
```
this seems better as a immutable constant calculated once at compile instead of calculated on each call

rackstar replied:
@MilGard91 if we want to make very minimal changes we can hardcode the hash value instead:
```solidity
// keccak256("RiQuote(uint256 coverId,uint24 productId,...)")
bytes32 internal constant RI_QUOTE_TYPEHASH =  0x...; 
```

rackstar replied:
just realized hardcodong '0x...' is the same as below. so we should just go with below as its more readable: 

```solidity
bytes32 internal constant RI_QUOTE_TYPEHASH = keccak256("RiQuote(uint256 coverId,uint24 productId,...)")
```

rackstar commented on contracts/modules/cover/Cover.sol:
we should drop this if we want NXM as paymentAsset. `_buyCover` function already handles the payment validation:
```
require(params.paymentAsset == params.coverAsset || params.paymentAsset == NXM_ASSET_ID, InvalidPaymentAsset());
```

MilGard91 committed:
feat: add getter for ri provider config

MilGard91 committed:
fix: set BUY_COVER_WITH_RI_TYPEHASH as a constant

MilGard91 committed:
fix: release docs

MilGard91 committed:
test: ri fork test

MilGard91 committed:
fix: bring back typedef AddressLike

MilGard91 committed:
test: add buyCoverWithRi to bft

MilGard91 committed:
fix: set package-lock.json lockfileVersion to 3

# Conflicts:
#	package-lock.json
#	package.json

# Conflicts:
#	package-lock.json

shark0der committed:
refactor: extract encodeRiData and add RiData JSDoc types
